### PR TITLE
Fixes for Coverity unchecked return values problems

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
Ping for review

File: test/bad_dtls_test.c [hunk 1]
```diff
@@ -329,11 +330,12 @@ static int send_record(BIO *rbio, unsigned char type, uint64_t seqnr,
     } while (len % 16);
 
     /* Generate IV, and encrypt */
-    RAND_bytes(iv, sizeof(iv));
-    enc_ctx = EVP_CIPHER_CTX_new();
-    EVP_CipherInit_ex(enc_ctx, EVP_aes_128_cbc(), NULL, enc_key, iv, 1);
-    EVP_Cipher(enc_ctx, enc, enc, len);
-    EVP_CIPHER_CTX_free(enc_ctx);
+    if (!TEST_true(RAND_bytes(iv, sizeof(iv)))
```
Comment (t8m) [inline test/bad_dtls_test.c:333]: Above there are unchecked calls to EVP_MAC_... functions I suppose they'll be marked by Coverity later as well.

File: test/exptest.c [hunk 2]
```diff
@@ -144,17 +144,22 @@ static int test_mod_exp(int round)
         || !TEST_ptr(m = BN_new()))
         goto err;
 
-    RAND_bytes(&c, 1);
+    if (!TEST_true(RAND_bytes(&c, 1)))
+        goto err;
     c = (c % BN_BITS) - BN_BITS2;
     BN_rand(a, NUM_BITS + c, BN_RAND_TOP_ONE, BN_RAND_BOTTOM_ANY);
```
Comment (t8m) [inline test/exptest.c:150]: I suppose this BN_rand() call and the one below should be checked as well.

**Review by t8m — CHANGES_REQUESTED** [review]

File: test/bad_dtls_test.c (hunk 1 above)
Comment (paulidale) [reply test/bad_dtls_test.c:333]: If enough callers check the result, they will be flagged.

I'll add some extra checks.

File: test/exptest.c (hunk 2 above)
Comment (paulidale) [reply test/exptest.c:150]: Same here.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Rebasing to master to clear the merge conflict.  Can't merge to master due to locked repo.

**Comment by paulidale** [comment]
Thanks for the comments.  Merged to master.

**Pull request closed without merging**